pycbc/types/timeseries.py

```diff
@@ -856,19 +856,39 @@ def add_into(self, other):
         other = other.copy()
         dt = float((other.start_time - self.start_time) * self.sample_rate)
 
+        # This coaligns other to the time stepping of self
         if not dt.is_integer():
             diff = (dt - _numpy.floor(dt)) * self.delta_t
+
+            # insert zeros at end
             other.resize(len(other) + (len(other) + 1) % 2 + 1)
+
+            # fd shift to the right
             other = other.cyclic_time_shift(diff)
 
-        ts = self.copy()
-        start = max(other.start_time, self.start_time)
-        end = min(other.end_time, self.end_time)
+        # get indices of other with respect to self
+        # this is already an integer to floating point precission
+        left = float(other.start_time - self.start_time) * self.sample_rate
+        left = int(round(left))
```

Comment:
Should be floor here, right?

Reply:
@lpsinger No here no. The time stamps should already be aligned so they'll differ by exactly an integer up to floating point precision. If you do floor, then you can have an off-by-one issue if the floating point number just slightly drifts below the correct integer. (i.e. 1 - 1e7)

Reply:
I see.